features/environment.py

```diff
@@ -846,6 +865,13 @@ def create_and_set_output_directory(self, feature_name):
             shutil.rmtree(feature_dir)
         os.makedirs(feature_dir)
         self._output_dir = feature_dir
+    
+    def read_patroni_log(self, type: str, node: str) -> 'list[str]':
+        try:
+            with open(str(os.path.join(self._output_dir or '', "patroni_" + node + ".log"))) as f:
+                return [line for line in f.readlines() if line[24:24+len(type)] == type]
+        except IOError:
+            return []
```

Comment:
Please move this method to the `PatroniController` class.
It could be accessed using https://github.com/zalando/patroni/pull/2724/files (`read_patroni_log` should be added to the list).

Reply:
Of course, arguments should be reordered (`name` comes first).